simSPI/fov.py

```diff
@@ -0,0 +1,249 @@
+
+    Returns
+    -------
+    traj_small.xyz : ndarray
+        Particle coordinates from .pdb file.
+    """
+    traj = md.load(filename)
+    atom_indices = traj.topology.select(atom_selection)
+    traj_small = traj.atom_slice(atom_indices)
+    coordinates = traj_small.xyz
+
+    return coordinates
+
+
+def micrograph2particles(
+    micrograph, optics_params, detector_params, pdb_file=None, dmax=30, pad=5.0
+):
+    """Extract particles from given micrograph.
+
+    Parameters
+    ----------
+    micrograph : ndarray
+        Particle micrograph to extract from.
+    optics_params : list
+        List of sim parameters pertaining to microscope settings.
+    detector_params : list
+        List of sim parameters pertaining to detector settings.
+    pdb_file : str
+        Relative path to .pdb file.
+    dmax : int
+        Predefined maximum dimension of molecule.
+    pad : int
+        Amount of padding for each particle box.
+
+    Returns
+    -------
+    particles : ndarray
+        Picked and sliced particle data from micrograph.
+    """
+    fov_Lx, fov_Ly, boxsize = get_fov(
+        optics_params, detector_params, pdb_file=pdb_file, dmax=dmax, pad=pad
+    )
+    fov_Nx = np.floor(fov_Lx / boxsize)
+    fov_Ny = np.floor(fov_Ly / boxsize)
+    pixel_size = (fov_Lx / micrograph.shape[1] + fov_Ly / micrograph.shape[0]) / 2.0
+    n_boxsize = np.int(boxsize / pixel_size)
+    x_pixels = np.int(fov_Nx * n_boxsize)
+    y_pixels = np.int(fov_Ny * n_boxsize)
+    data = micrograph[0:y_pixels, 0:x_pixels]
+    particles = slice_and_stack(data, n_boxsize=n_boxsize)
+
+    return particles
+
+
+def slice_and_stack(data, n_boxsize=256, n_ovl=0):
+    """Convert a 2D numpy array into a 3D numpy array.
+
+    Parameters
+    ----------
+    data : ndarray
+        2D array to stack.
+    n_boxsize : int
+        Boxsize.
+    n_ovl : int
+        Overlap.
+
+    Returns
+    -------
+    data_stack : ndarray
+        Array containing sliced particle data.
+    """
+    log = logging.getLogger()
+
+    if n_ovl == 0:
+        data_stack = blockshaped(data, n_boxsize, n_boxsize)
+    else:
+        n_split = math.floor((data.shape[0] - 2 * n_ovl) / n_boxsize)
+        n_dilat = n_boxsize + 2 * n_ovl
+        data_stack = np.zeros((n_split * n_split, n_dilat, n_dilat))
+        log.info("Array dimensions: {}".format(data_stack.shape))
+        i_stack = 0
+        for i in np.arange(n_split):
+            for j in np.arange(n_split):
+                istart = i * n_boxsize
+                istop = istart + n_dilat
+                jstart = j * n_boxsize
+                jstop = jstart + n_dilat
+                rows = np.arange(istart, istop)
+                columns = np.arange(jstart, jstop)
+                data_tmp = data[np.ix_(rows, columns)]
+                data_stack[i_stack, ...] = data_tmp[np.newaxis, ...]
+                i_stack += 1
+
+    return data_stack
+
+
+def blockshaped(arr, nrows, ncols):
```

Comment:
For consistency of variable names: `arr` -> `data`?

Reply:
How is this function different from just doing `arr.reshape(-1, nrows, ncols)`? What is the expected input shape of the arr?

Reply:
> How is this function different from just doing arr.reshape(-1, nrows, ncols)? What is the expected input shape of the arr?

the swapaxes moves data between different axes.